### PR TITLE
Bio-Formats decoupling: developer docs

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/phTsSDsj/25-documentation-changes-scoping

These changes should modify the developers documentation to cover the changes brought into the mainline with https://github.com/openmicroscopy/openmicroscopy/pull/3931

### CODE REVIEW

File: omero/developers/build-system.txt [hunk 1]
```diff
@@ -134,24 +135,26 @@ OMERO build system defines and uses two kinds of caches:
 :ivydoc:`repository <resolver/ibiblio.html>`, a
 :ivydoc:`generic URL <resolver/url.html>` or from the
 :ivydoc:`local file system <resolver/filesystem.html>`.
+Since OMERO 5.1.3, the rremote repository resolvers are set up to resolve
```
Comment (hflynn) [inline omero/developers/build-system.txt]: rremote typo

File: omero/developers/build-system.txt [hunk 2]
```diff
@@ -134,24 +135,26 @@ OMERO build system defines and uses two kinds of caches:
...
 resolves the following locations in order:
 
-#. :file:`target/repository`
-#. :file:`lib/repository`
-#. :file:`target/test-repository`
-#. :file:`~/.m2/repository`
-#. `Maven central repository <http://central.sonatype.org/>`_
-#. `OME artifactory <http://artifacts.openmicroscopy.org>`_
-#. `SpringSource repository <http://repository.springsource.com/>`_
+#. the :file:`target/repository` which contains most artifacts published by the
```
Comment (hflynn) [inline omero/developers/build-system.txt]: don't really need 'the' in front of all these list items

**Comment by mtbc** [comment]
The bulk of the PR is staged at http://www.openmicroscopy.org/site/support/omero5.1-staging/developers/build-system.html.

**Comment by mtbc** [comment]
Good to merge.

**Pull request merged**